src/components/Presenter.js

```diff
@@ -0,0 +1,358 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import './Presenter.css';
+
+const PresenterStatus = {
+  INIT: 0,
+  LOADED: 1,
+  PLAYING: 2,
+  PAUSED: 3,
+  STOPPED: 4,
+};
+
+const ItemStatus = {
+  INIT: 0,
+  LOADED: 1,
+  ENTERING: 2,
+  ENTERED: 3,
+  LEAVING: 4,
+  LEFT: 5,
+};
+
+export const PresenterDirection = {
+  RIGHT: 2,
+  LEFT: 3,
+};
+
+const TransitionHorizontalSlide = {
+  // STYLE TO DISPLAY THE SLIDE
+  normal: {
+    opacity: 1,
+    left: 0,
+    right: 0,
+  },
+
+  // STYLE TO START SLIDE
+  enter: {
+    opacity: 0,
+    left: '-100%',
+    right: '100%',
+  },
+
+  // STYLE TO FINISH SLIDE
+  left: {
+    opacity: 0,
+    left: '100%',
+    right: '-100%',
+  },
+};
+
+class PresenterItem extends React.Component {
+  render() {
+    const { item, transitionDuration, enterStyle, stateStyle } = this.props;
+    if (item) {
+      return (
+        <div style={stateStyle} className="item">
+          <img src={item.image} />
+          <div className="text">
+            <p>{item.text}</p>
+          </div>
+        </div>
+      );
+    } else return null;
+  }
+}
+
+class Presenter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: -1,
+      status: PresenterStatus.INIT,
+      current: null,
+      direction: props.direction,
+    };
+    this.timerChange = null;
+
+    this.setupNode = this.setupNode.bind(this);
```

Comment:
I think binding is not needed on any of this functions if refactores as `<name> = () => `

Reply:
Exactly. I forgot, I was passing some other parameter before. I am going to keep the binding here and remove the arrow funcion in the callback.